### PR TITLE
Remove jvm annotations, make NativeRandom internal

### PR DESCRIPTION
**Pull request by ilya-g**

### CODE REVIEW

File: runtime/src/main/kotlin/kotlin/random/Random.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ package kotlin.random
 import kotlin.native.concurrent.AtomicLong
 import kotlin.system.getTimeNanos
 
-public abstract class NativeRandom {
+internal class NativeRandom {
```
Comment (ilya-g) [inline runtime/src/main/kotlin/kotlin/random/Random.kt]: Should we keep default implementation as companion object or we can make it just a top-level object?

File: runtime/src/main/kotlin/kotlin/random/Random.kt [hunk 2]
```diff
@@ -28,39 +28,12 @@ public abstract class NativeRandom {
...
         }
 
-        /**
-         * Returns a pseudo-random Int value between 0 and specified value (exclusive)
-         */
-        override fun nextInt(bound: Int): Int {
-            if (bound <= 0) throw IllegalArgumentException("Incorrect bound: $bound")
-
-            if (bound and (bound - 1) == 0) {
-                return ((bound * (nextInt() ushr 1).toLong()) ushr 31).toInt();
```
Comment (ilya-g) [inline runtime/src/main/kotlin/kotlin/random/Random.kt]: I don't think this marginal implementation difference deserves its own reimplementation that has to be maintained and tested.

File: runtime/src/main/kotlin/kotlin/random/Random.kt [hunk 3]
```diff
@@ -28,39 +28,12 @@ public abstract class NativeRandom {
...
-                var r = nextInt() ushr 1
-                m = r % bound
-            } while (r - m + (bound - 1) < 0)
-            return m
-        }
-
-        /**
-         * Returns a pseudo-random Long number.
-         */
-        override fun nextLong(): Long = (nextInt().toLong() shl 32) + nextInt().toLong()
```
Comment (olonho) [inline runtime/src/main/kotlin/kotlin/random/Random.kt]: Why you removed this one?

File: runtime/src/main/kotlin/kotlin/random/Random.kt (hunk 2 above)
Comment (olonho) [reply runtime/src/main/kotlin/kotlin/random/Random.kt]: Could you please clarify how it shall affect clients of RNG?

File: runtime/src/main/kotlin/kotlin/random/Random.kt (hunk 3 above)
Comment (ilya-g) [reply runtime/src/main/kotlin/kotlin/random/Random.kt]: The same implementation is inherited from the base class.

File: runtime/src/main/kotlin/kotlin/random/Random.kt (hunk 2 above)
Comment (ilya-g) [reply runtime/src/main/kotlin/kotlin/random/Random.kt]: Perhaps getting a random number in [0..bound) range where bound is 2^n will be not so fast as in JVM, until we make `fastLog2` (or `Int.numberOfLeadingZeroes` operation) really fast.

**Comment by PavelPunegov** [comment]
|`runtime/src/main/kotlin/kotlin/random/Random.kt:49`|
|---|
I would rather have this removed also as there are no need to have such seeding available in native stdlib. Tests that use it could be just moved to common stdlib tests or removed

File: runtime/src/main/kotlin/kotlin/random/Random.kt (hunk 1 above)
Comment (PavelPunegov) [reply runtime/src/main/kotlin/kotlin/random/Random.kt]: I think we can do make it a top level

File: runtime/src/main/kotlin/kotlin/random/Random.kt (hunk 2 above)
Comment (PavelPunegov) [reply runtime/src/main/kotlin/kotlin/random/Random.kt]: we have https://github.com/JetBrains/kotlin-native/blob/79755d8fbfaf495e642fc41348b99029c4f6fbde/runtime/src/main/kotlin/kotlin/native/internal/HexStringParser.kt#L272
It looks the same as in JVM so we can make `Int.numberOfLeadingZeroes` be as fast as in the JVM (without taking jit into account)

**Comment by PavelPunegov** [comment]
Than this method should be explicitly intrinsified

**Comment by PavelPunegov** [comment]
Then this method should be explicitly intrinsified

**Pull request merged**